src/coloring/edgeColoring.cpp

```
@@ -29,112 +33,59 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
-        return V_to_id.at(v);
-    } catch (...) {
-        throw std::make_pair(
-            std::string("INTERNAL: something went wrong when getting the vertex id"),
-            std::string(__PGR_PRETTY_FUNCTION__));
+    size_t i = 0;
+    boost::graph_traits<B_G>::edge_iterator ei, ei_end;
+    for (boost::tie(ei, ei_end) = boost::edges(g.graph); ei != ei_end; ++ei, ++i) {
+        results[i].d1.id = static_cast<int64_t>(g[*ei].id);
+
+        /**
+         * There is a problem with boost:
```

Cppcheck (2.20.0)

[performance] Function parameter 'data_edges' should be passed by const reference.

(passedByValue)